nautobot/extras/plugins/__init__.py

```diff
@@ -172,6 +176,18 @@ def ready(self):
                 register_secrets_provider(secrets_provider)
             self.features["secrets_providers"] = secrets_providers
 
+        # Register custom filtersets (if any)
+        filter_extensions = import_object(f"{self.__module__}.{self.filter_extensions}")
+        if filter_extensions is not None:
+            register_filter_extensions(filter_extensions, self.base_url)
```

Comment:
@glennmatthews the reason `base_url` was chosen was because `name` can be pretty long, since a lot of the plugins include `nautobot_` in the beginning of the name which makes them significantly long.

Ex: https://github.com/nautobot/nautobot-plugin-golden-config/blob/develop/nautobot_golden_config/__init__.py#L11

Looking at secrets provider though, do we force uniqueness for `base_url`?

Ex: https://github.com/nautobot/nautobot-plugin-secrets-providers/blob/develop/nautobot_secrets_providers/__init__.py#L22

Reply:
An undocumented stylistic pattern is for `base_url` to use hyphens for multi-word names. Example https://github.com/nautobot/nautobot-plugin-device-lifecycle-mgmt/blob/develop/nautobot_device_lifecycle_mgmt/__init__.py#L31

Reply:
My vote would be to go with the name of the plugin which is the name of the django app name. This sort of pattern (`app_label`.`model_name`.`field_name`) is very standard in django, rather than reinventing the wheel here.

Reply:
@lampwins In the context of extending existing model's FilterSet, we discussed forced namespacing of the `field_name` so as to not have to content with collision with core.

That would generate from my perspective:

`dcim.site.nautobot_golden_config_new-custom-field-name` if extending Site for example via Golden Config.

Reply:
Wouldn't that just be `nautobot_golden_config.new_custom_field_name`? Still seems better to me than `golden-config-new-custom-field-name`.